mongorestore/restore.go

```diff
@@ -207,7 +207,7 @@ func (restore *MongoRestore) RestoreIntent(intent *intents.Intent) Result {
 			indexes = metadata.Indexes
 			if restore.OutputOptions.PreserveUUID {
 				if metadata.UUID == "" {
-					return Result{Err: fmt.Errorf("--preserveUUID used but no UUID found in %v", intent.MetadataLocation)}
+					log.Logvf(log.Always, "--preserveUUID used but no UUID found in %v, generating new UUID for %v", intent.MetadataLocation, intent.Namespace())
```

Comment:
Good idea to generate a UUID for the null case! However, where is the UUID generated since it would be set as null in line 212?

Reply:
Moreover, we should probably add a test case that ensures that a UUID is correctly generated since this is not the default behavior (correct me if I'm wrong here) of mongorestore.

Reply:
Great question! The UUID for collections is generated by the server. 

If we have a zero-value UUID, then we create the collection with a `create` command. There's no way of passing a UUID to a `create` command, it will always generate a new one itself on MongoDB 3.6+. 

If we have a UUID value, then we use an `applyOps` command to create the collection. `applyOps` lets us bypass a lot of regular restrictions with commands so we can use it to manually set a UUID that the server has to use.

I think we don't need to check the UUIDs that eventually get created, since that's a server thing.

Reply:
Ahh okay, this makes sense! In that case, LGTM :)